src/sbvr-api/hooks.ts

```diff
@@ -31,23 +30,27 @@ export interface HookReq {
 	hooks?: InstantiatedHooks;
 	is?: (type: string | string[]) => string | false | null;
 }
-export interface HookArgs {
+export interface HookArgs<Vocab extends string = string> {
 	req: HookReq;
 	request: ODataRequest;
-	api: PinejsClient;
+	api: (typeof api)[Vocab];
 	tx?: Tx | undefined;
 }
 export type HookResponse = PromiseLike<any> | null | void;
 
-export interface Hooks {
-	PREPARSE?: (options: Omit<HookArgs, 'request' | 'api'>) => HookResponse;
-	POSTPARSE?: (options: HookArgs) => HookResponse;
-	PRERUN?: (options: HookArgs & { tx: Tx }) => HookResponse;
+export interface Hooks<Vocab extends string = string> {
```

Comment:
Couldn't this and the above be `export interface HookArgs<Vocab extends (keyof typeof api) = string>` ? I think all in all it is the same, but this is a bit more explicit for the reader on where the string should be coming from

Reply:
Maybe something like `type VocabKeys = (keyof typeof api)` and `Vocab extends VocabKeys` all over the place

Reply:
In theory yes it could be that, but `keyof typeof api` is `string` anyway because of the index signature, so in practice there is no difference